### PR TITLE
Replace deprecated data template_file with templatefile function

### LINKED ISSUE #87: Deprecated template provider should be removed from module
## Describe the Bug
Template provider used in this module is long deprecated, it should be removed. It does not exist for Mac M1 machines, so you either need to use workarounds or it would not work.

https://github.com/hashicorp/terraform-provider-template/issues/85

## Expected Behavior
Module should work on all platforms supported on TF.

## Steps to Reproduce
Do whatever

## Screenshots
```
Initializing modules...
Downloading registry.terraform.io/cloudposse/ec2-autoscale-group/aws 0.27.0 for autoscale_group...
- autoscale_group in .terraform/modules/autoscale_group
Downloading registry.terraform.io/cloudposse/label/null 0.24.1 for autoscale_group.this...
- autoscale_group.this in .terraform/modules/autoscale_group.this
Downloading registry.terraform.io/cloudposse/label/null 0.25.0 for label...
- label in .terraform/modules/label
Downloading registry.terraform.io/cloudposse/security-group/aws 0.3.3 for security_group...
- security_group in .terraform/modules/security_group
Downloading registry.terraform.io/cloudposse/label/null 0.25.0 for security_group.this...
- security_group.this in .terraform/modules/security_group.this
Downloading registry.terraform.io/cloudposse/label/null 0.25.0 for this...
- this in .terraform/modules/this

Initializing the backend...

Successfully configured the backend "s3"! Terraform will automatically
use this backend unless the backend configuration changes.

Initializing provider plugins...
- Finding hashicorp/null versions matching ">= 2.0.0"...
- Finding integrations/github versions matching "~> 4.0"...
- Finding hashicorp/aws versions matching ">= 2.0.0, >= 3.63.0, < 4.0.0"...
- Finding hashicorp/template versions matching ">= 2.0.0"...
- Installing hashicorp/null v3.1.1...
- Installed hashicorp/null v3.1.1 (signed by HashiCorp)
- Using integrations/github v4.24.1 from the shared cache directory
- Using hashicorp/aws v3.75.1 from the shared cache directory
╷
│ Error: Incompatible provider version
│
│ Provider registry.terraform.io/hashicorp/template v2.2.0 does not have a
│ package available for your current platform, darwin_arm64.
│
│ Provider releases are separate from Terraform CLI releases, so not all
│ providers are available for all platforms. Other versions of this provider
│ may have different platforms supported.
╵

ERRO[0033] 1 error occurred:
	* exit status 1
```

## Environment (please complete the following information):

Anything that will help us triage the bug will help. Here are some ideas:
 - OS: Mac Os, Apple Silicon M1
 - Version: Monterey 12.3.1

### PR DESCRIPTION
**Pull request by carl-reverb**
## what
* Replace deprecated data template_file with templatefile function

## why
* Users of arm architecture computers cannot use deprecated template provider

## references
* closes #87

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -98,29 +108,14 @@ data "aws_ami" "eks_worker" {
...
-  }
-}
-
 data "aws_iam_instance_profile" "default" {
   count = local.enabled && var.use_existing_aws_iam_instance_profile ? 1 : 0
   name  = var.aws_iam_instance_profile_name
 }
 
-module "autoscale_group" {
+module "autoscale_roup" {
```
Comment (nitrocode) [inline main.tf:116]:
```suggestion
module "autoscale_group" {
```

**Comment by nitrocode** [comment]
Please see comment and run `make pr/readme` locally to fix the `Auto Format` failure.

**Comment by nitrocode** [comment]
It seems that the bot may not be able to update the PR. Could you allow the bot to make changes ? Usually personal forks allow this out of the box and only company specific forks prevent this.

**Comment by carl-reverb** [comment]
I have added the bot as a contributor but its owner must accept the invitation so that I may assign the proper permission.

**Comment by nitrocode** [comment]
cc: @aknysh @Gowiem for visibility. Not sure what to do here. Usually we do not need to accept the invite. Perhaps there is another workaround ? Usually open source forks of our repos that are non-company specific orgs work out of the box.

Also, please try pushing a new commit to rerun the auto format check.

**Comment by nitrocode** [comment]
We could. I think another way is if you could run `make pr/auto-format` locally and push up the files. If you have time, could you do this ?

**Review by Nuru — APPROVED** [review]

**Pull request merged**